model-server-openapi/specifications/model-server.yaml

```diff
@@ -13,481 +13,700 @@ info:
   description: "modelix public API"
   version: "1.0.0"
 servers:
-  - url: '/'
+  - url: '/v2'
     description: model-server
+
+tags:
+  - name: IDs
+  - name: repository management
+  - name: branch management
+  - name: versions
+    description: TODO what is this?
+  - name: delta access
+    description: Used for incrementally updating a branch or requesting data from a known state.
+  - name: node access
+    description: TODO this is a rough sketch
+  - name: object access
+    description: |-
+      Direct access to the key-value store.
+
+      TODO:
+      * What about access to KVs without repo context? Security issue? Removed for now
+  - name: ModelQl access
+  - name: WebSocket access
+
 paths:
-  /v2/server-id:
+
+  /server-id:
     get:
       operationId: getServerId
+      summary: returns the static ID of the server
+      tags:
+        - IDs
       responses:
         "200":
-          $ref: '#/components/responses/200'
-  /v2/user-id:
+          description: Successful response for requesting the server ID.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ServerId"
+        default:
+          $ref: "#/components/responses/Problem"
+  /client-id:
+    post:
+      operationId: generateClientId
+      summary: Generate a new and unique client ID
+      tags:
+        - IDs
+      responses:
+        "201":
+          description: A new client ID was created successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ClientId"
+        default:
+          $ref: "#/components/responses/Problem"
+
+  /repositories:
     get:
-      operationId: getUserId
+      operationId: getRepositories
+      summary: Returns all existing repositories
+      tags:
+        - repository management
       responses:
         "200":
-          $ref: '#/components/responses/200'
-  /v2/generate-client-id:
+          description: A new client ID was created successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/RepositoryList"
+        default:
+          $ref: "#/components/responses/Problem"
     post:
-      operationId: postGenerateClientId
+      operationId: initializeRepository
+      summary: Creates a new repository
```

Comment:
For me it's not intuitive that this endpoint creates a new repository. Also specifying the repository name in the JSON body while all other endpoint specify it in the URL seems inconsistent. I would prefer a `POST /repositories/{repository}` endpoint.

Reply:
> POST requests are idiomatically used to create single resources on a collection resource endpoint,

https://opensource.zalando.com/restful-api-guidelines/#post